[Dashboard] Display duration in milliseconds if under 1 second.

coltwood93 opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this solves. -->
On the Ray Dashboard, a duration less than 1 second is displayed as "0s", which is not accurate.

## Related issue number

<!-- For example: "Closes #1234" -->
Closes #49010 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

coltwood93 committed:
Display duration in milliseconds if under 1 second.

Signed-off-by: Colton Woodruff <coltwood93@gmail.com>

coltwood93 committed:
Merge branch 'master' into master

coltwood93 committed:
Merge branch 'master' into master

coltwood93 committed:
Merge branch 'master' into master

MortalHappiness committed:
[Refactor][GCS] Merge RedisAsioClient into RedisAsyncContext (#49000)

Signed-off-by: Chi-Sheng Liu <chishengliu@chishengliu.com>

dentiny committed:
Fix linting (#49092)

Signed-off-by: hjiang <dentinyhao@gmail.com>

dentiny committed:
Another cleanup for test main (#49093)

Our usage for gtest and dependency is incorrect, in short, we shouldn't
declare `@com_google_googletest//:gtest_main` as dependency and
**meanwhile** have main function defined in the test.

One weird behavior I noticed is, `SetUp` overriden function will be
invoked twice somehow.
Similar to https://github.com/ray-project/ray/pull/48892

Signed-off-by: hjiang <dentinyhao@gmail.com>

simonsays1980 committed:
[RLlib] Move all new stack Offline RL classes to PublicAlpha, stability=alpha. (#49101)

dentiny committed:
[core] Cleanup clang-format off in prod code (#49090)

Signed-off-by: hjiang <dentinyhao@gmail.com>

KPostOffice committed:
[data] move _munge_conflict to parent class (#46668)

## Why are these changes needed?

PandaDataAccessor does not have function _munge_conflict

## Related issue number

closes: #46275

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: Kevin <kpostlet@redhat.com>
Co-authored-by: Cheng Su <scnju13@gmail.com>

simonsays1980 committed:
[RLlib; Offline RL] Enable offline training without specifying an environment. (#49041)

kevin85421 committed:
[core][compiled-graphs] remove `actor_task_count` (#49086)

Signed-off-by: kaihsun <kaihsun@anyscale.com>

alanwguo commented on python/ray/dashboard/client/src/common/DurationText/DurationText.tsx:
```diff
@@ -27,7 +27,9 @@ export const DurationText = ({ startTime, endTime }: DurationTextProps) => {
   let durationText: string;
   let refreshInterval = 1000;
-  if (duration.asMinutes() < 1) {
+  if (duration.asSeconds() < 1) {
+    durationText = duration.format("SSS[ms]");
+  } else if (duration.asMinutes() < 1) {
```
For numbers less than 1 minute, can we include milliseconds as well?

alanwguo replied:
also, can you add a test case in `DurationText.component.test.tsx`?

coltwood93 replied:
> For numbers less than 1 minute, can we include milliseconds as well?

Sure, I can add that! I'll add it momentarily.

> also, can you add a test case in DurationText.component.test.tsx?

Yes, I'll take a look and see what I can come up with. Thanks for the review!

alanwguo reviewed (commented):
Thanks for picking this up!

ruisearch42 committed:
[core][compiled graph] Fix execution schedule visualization (#49055)

When we simplified the node visualization to exclude actor information, the node visualization ID could conflict with other nodes when all other information (e.g., operation type, index in the schedule) is the same. This PR fixes the issue by using a visualization ID that includes actor information, but the label continues to exclude actor information for conciseness.

bveeramani committed:
[Data] Sort Data release tests (#49106)

This PR groups the Data release tests into sections. There aren't any actual changes to the tests themselves or their setups.

---------

Signed-off-by: Balaji Veeramani <bveeramani@berkeley.edu>

Superskyyy committed:
Move num_cpu note to resources page to avoid confusion (#48871)

closes: https://github.com/ray-project/ray/issues/48867

Signed-off-by: Superskyyy <yihaochen@apache.org>

pcmoritz committed:
[KubeRay] Add instructions on how to use Kustomize (#49085)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Currently there are some docs in
https://ray-project.github.io/kuberay/deploy/installation/ but they are
not complete and didn't work for me, instead add working instructions
and also show how to delete stuff.

This might be suboptimal -- open to whether we want to merge it. If we
do, we need to figure out how to name the cluster `raycluster-kuberay`
instead of `raycluster-sample` so the rest of the instructions still
work.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

dentiny committed:
[core] [7/N] Cleanup gcs server shared pointer (#49089)

Cleanup shared pointer and use unique pointer for clear memory ownership
and less error prune.

---------

Signed-off-by: hjiang <dentinyhao@gmail.com>

dentiny committed:
[core] Remove useless function (#49110)

Remove a useless function:
- `BroadcastRaySyncMessage` is just a wrapper around `BroadcastMessage`,
doing nothing else
- The comment is confusing also, I don't see anything dangerous which
breaks the state machine

Signed-off-by: hjiang <dentinyhao@gmail.com>

rkooo567 committed:
[Core][Fix] Fix "No error is raised when the "bind" args does not match the function signature" (#47773)

bind doesn't check if the arg matches the signature. Fortunately, we already have the logic in .remote that checks it, and we just use the same mechanism. This also provides very nice error message;

raulchen committed:
[data] Fix DataContext sealing for multiple datasets.  (#49096)

## Why are these changes needed?

When users using multiple datasets and want to set different DataContext
configurations.
The recommended way is to set `DataContext.get_current()` before
creating a Dataset. The DataContext is supposed to be captured and
sealed by a Dataset when it's created. For example:

```python
                import ray

                context = ray.data.DataContext.get_current()

                context.target_max_block_size = 100 * 1024 ** 2
                ds1 = ray.data.range(1)
                context.target_max_block_size = 1 * 1024 ** 2
                ds2 = ray.data.range(1)

                # ds1's target_max_block_size will be 100MB
                ds1.take_all()
                # ds2's target_max_block_size will be 1MB
                ds2.take_all()
```

However in Ray Data internal code, `DataContext.get_current()` has been
widely used in an incorrect way. This PR fixes most outstanding issues
(but not all), by explicitly passing around the captured DataContext
object as an argument to each component.

## Related issue number
https://github.com/ray-project/ray/issues/41573

---------

Signed-off-by: Hao Chen <chenh1024@gmail.com>

dentiny committed:
Fix linting (#49121)

Signed-off-by: hjiang <dentinyhao@gmail.com>

dentiny committed:
[Core] Suppress unused function warning for 3rd party libraries (#49018)

Signed-off-by: hjiang <dentinyhao@gmail.com>

coltwood93 committed:
added milliseconds to durations under 1 minute

Signed-off-by: Colton Woodruff <coltwood93@gmail.com>

coltwood93 committed:
added/modified tests to support milliseconds in duration

Signed-off-by: Colton Woodruff <coltwood93@gmail.com>

coltwood93 commented:
I mistakenly committed without signing, so I'm trying to figure out how to correct that...

coltwood93 committed:
Merge branch 'master' into master

coltwood93 commented:
I think I just blew up this PR. Going to close and open a fresh one to resolve issue #49010
